src/System.Private.CoreLib/shared/System/Diagnostics/Tracing/BaseCounter.cs

```diff
@@ -0,0 +1,102 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Collections;
+using System.Collections.Generic;
+using System.Threading;
+#if ES_BUILD_PCL
+    using System.Threading.Tasks;
+#endif
+
+#if ES_BUILD_STANDALONE
+namespace Microsoft.Diagnostics.Tracing
+#else
+namespace System.Diagnostics.Tracing
+#endif
+{
+    /// <summary>
+    /// BaseCounter is an abstract class that serves as the parent class for various Counter* classes, 
+    /// namely EventCounter, PollingCounter, IncrementingEventCounter, and IncrementingPollingCounter.
+    /// </summary>
+    public abstract class BaseCounter : IDisposable
```

Comment:
Is there API proposal issue in CoreFX to add these new public APIs?

Reply:
Based on conversation with @vancem today, I will merge this PR as it is. I'll be filing an API review in CoreFX for PollingCounter, IncrementingEventCounter, and IncrementingPollingCounter later today.

cc @terrajobst for visibility.